test/test_gz.c

```diff
@@ -0,0 +1,189 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "test.h"
+#include "test_utils.h"
+
+#define DATALEN 1024*1024 // 1MB
+#define BUFLEN	8192
+
+//const char filename[14] = "test_gzfile.gz";
+
+int test_nx_gzwrite(Byte *src, int size, char *filename)
+{
+	gzFile file;
+	int rc;
+
+	file = nx_gzopen(filename,"wb");
+	if (file == NULL) {
+		printf("*** failed gzopen!\n");
+		return TEST_ERROR;
+	}
+
+	rc = nx_gzwrite(file, src, size);
+
+	if (rc != size) {
+		printf("*** failed: gzwrite returned %d, size is %d\n", rc,
+			size);
+		return TEST_ERROR;
+	}
+
+	if (nx_gzclose(file) != Z_OK) {
+		printf("*** failed gzclose!\n");
+		return TEST_ERROR;
+	}
+
+	return TEST_OK;
+}
+
+int test_nx_gzread(Byte *src, Byte *dest, int src_size, char *filename)
+{
+	gzFile file;
+	int rc;
+
+	file = nx_gzopen(filename,"rb");
+	if (file == NULL) {
+		printf("*** failed gzopen!\n");
+		return TEST_ERROR;
+	}
+
+	rc = nx_gzread(file, dest, src_size);
+
+	if (rc <= 0) {
+		printf("*** failed: gzread returned %d\n", rc);
+		return TEST_ERROR;
+	}
+
+	if (nx_gzclose(file) != Z_OK) {
+		printf("*** failed gzclose!\n");
+		return TEST_ERROR;
+	}
+
+	if (compare_data(src, dest, src_size))
+		return TEST_ERROR;
+
+	return TEST_OK;
+}
+
+int test_gzwrite(Byte *src, int size, char *filename)
+{
+	gzFile file;
+	Byte *cur = src;
+	int rc;
+	char buf[BUFLEN];
```

Comment:
Why is this buffer used to read the data from `src` here? Can't we pass `src` directly to `gzwrite` like in `test_nx_gzwrite`?

Reply:
`test_nx_gzwrite` writes all data in one `gzwrite` call, buf if we pass `src` to `gzwrite` here each loop iteration will overwrite the data from the previous iteration.

Reply:
Anyway, we don't really need `buf` as we can pass `cur` to `gzwrite`.
I'll change that, thanks!